src/core/grpc_service.proto

```diff
@@ -250,6 +256,100 @@ message ControlResponse
   RequestStatus request_status = 1;
 }
 
+//@@.. cpp:var:: message InferSharedMemory
+//@@
+//@@   The meta-data for the shared memory from which to read the input
+//@@   data and/or write the output data.
+//@@
+message InferSharedMemory
+{
+  //@@  .. cpp:var:: string shm_key
+  //@@
+  //@@     The name of the shared memory region that holds the input data
+  //@@     (or where the output data should be written).
+  //@@
+  string shm_key = 1;
+
+  //@@  .. cpp:var:: uint64 offset
+  //@@
+  //@@     The offset from the start of the shared memory region.
+  //@@     start = offset, end = offset + size;
+  //@@
+  uint64 offset = 2;
+
+  //@@  .. cpp:var:: uint64 byte_size
+  //@@
+  //@@     Size of the memory block, in bytes.
+  //@@
+  uint64 byte_size = 3;
+}
+
+//@@
+//@@.. cpp:var:: message ShmControlRequest
+//@@
+//@@   Request message for managing registered shared memory regions in TRTIS.
+//@@
+message ShmControlRequest
+{
+  //@@  .. cpp:enum:: Type
+  //@@
+  //@@     Types of control operation
+  //@@
+  enum Type {
+    //@@    .. cpp:enumerator:: Type::UNREGISTER = 0
+    //@@
+    //@@       To unregister the specified shared memory region.
+    //@@
+    UNREGISTER = 0;
+
+    //@@    .. cpp:enumerator:: Type::REGISTER = 1
+    //@@
+    //@@       To register the specified shared memory region. If used the
+    //@@       InferSharedMemory message and all its fields are required.
+    //@@
+    REGISTER = 1;
+  }
+
+  //@@
+  //@@  .. cpp:var:: string model_name
+  //@@
+  //@@     The target shared memory region name (different from the shm_key)
+  //@@     that the user will use to refer to the specified region.
+  //@@
+  string shm_name = 1;
```

Comment:
I would put this inside InferSharedMemoryRegion. The reason is that we need a "status" API as well that returns all the shared memory regions and the response of that can just be a repeated of InferSharedMemoryRegion.